src/node_worker.cc

```diff
@@ -453,10 +453,12 @@ void Worker::JoinThread() {
     HandleScope handle_scope(env()->isolate());
     Context::Scope context_scope(env()->context());
 
+    // TODO(qard): Do we need to move this, or can we just drop it entirely?
+    // See: https://github.com/nodejs/node/issues/56020
     // Reset the parent port as we're closing it now anyway.
-    object()->Set(env()->context(),
-                  env()->message_port_string(),
-                  Undefined(env()->isolate())).Check();
+    // object()->Set(env()->context(),
+    //               env()->message_port_string(),
+    //               Undefined(env()->isolate())).Check();
```

Comment:
I would think we want to move it, especially if the value is pointing to an API object that is going away. Alternatively, we could wrap it in a check to see if the isolate is still viable? Not sure... @joyeecheung @addaleax any thoughts?

Reply:
I had initially attempted to move the field clear elsewhere, but it didn't work. I opted for just commenting out the line for the moment as, at least from the comment, it sounds like it's not actually required. I don't know the code here well enough to be sure though. It fixes the crash when applied to 20.x though, so would presumably fix it here too, assuming it's still possible, which I think is _likely_, but I don't have definitive proof. 🤔

Reply:
I am a bit surprised that the `MakeCallback()` call below is somehow fine even though `object()->Set()` is not okay...Shouldn't we skip them all if the env is already shut down here? Though I am not sure what's the right flag to use (`env->set_can_call_into_js()`?), @addaleax may know it better.

Reply:
MakeCallback has its own safety bailouts, so probably just hits that.